src/ui/home/components/claimDetails.tsx

```diff
@@ -1,40 +1,109 @@
+  state = {
+    line_1: '',
+    line_2: '',
+  }
 
+  componentWillMount() {
+    // TODO: adjust for multiline when enabled
+    const { value } = this.props.selectedClaim.claims[0]
+    const claimField = this.props.selectedClaim.type[1]
+    if (value && (claimField.toString() === 'ProofOfNameCredential')) {
+      const fullName = value.split(' ')
+      this.setState({
+        line_1: fullName[0],
+        line_2: fullName[1] ? fullName[1] : ''
+      })
+    } else if (value) {
+      this.setState({ line_1: value })
+    }
+  }
 
+  private handleFieldInput = (fieldValue: string, fieldName: string) => {
+    this.setState({[fieldName]: fieldValue})
+  }
 
-export const ClaimDetails : React.SFC<Props> = (props) => {
-  return (
-    <Container>
-      <Block>
-        <CenteredText
-          style={JolocomTheme.textStyles.light.subheader}
-          msg={props.typeClaimDetails} />
-
-      {/* <TextInput
-        placeholder={props.typeClaimDetails}
-        placeholderTextColor={'grey'}
-        style={{height: 40, width: 288}}
-      /> */}
-
+  private onSubmit = (claimsItem: DecoratedClaims) => {
+    claimsItem.claims[0].value = (claimsItem.type[1] === 'ProofOfNameCredential') ? this.prepareNameValue() : this.state.line_1
+    this.props.saveClaim(claimsItem)
+  }
 
+  private prepareNameValue = () => {
+    const { line_1, line_2 } = this.state
+    return line_1 && line_2 ? line_1 + " " + line_2 : line_1 + line_2
+  }
 
-      {/* <Block>
-        <TextField
-          label='Phone number'
-          value={'test'} /> */}
+  private renderInputFields = (displayName: string) => {
+    const { line_1, line_2 } = this.state
+    switch(displayName) {
+      case ('Name'):
+        return (
+          <Block>
+            <TextInputField
+              fieldName={ 'line_1' }
+              fieldValue={ line_1 }
+              displayName={ 'First Name' }
+              handleFieldInput={ this.handleFieldInput }
+            />
+            <TextInputField
+              fieldName={ 'line_2' }
+              fieldValue={ line_2 }
+              displayName={ 'Last Name' }
+              handleFieldInput={ this.handleFieldInput }
+            />
+          </Block>
+        )
+      default:
+        return (
+          <TextInputField
+            fieldName={ 'line_1' }
+            fieldValue={ line_1 }
+            displayName={ displayName }
+            handleFieldInput={ this.handleFieldInput }
+          />
+        )
+    }
+  }
 
-      </Block>
-    </Container>
-  )
+  render() {
+    if (this.props.selectedClaim.claims.length === 0) {
+      return
+    }
+    const claimField = this.props.selectedClaim.displayName
+    return (
+      <Container>
+        <Block>
+          <CenteredText
+            style={ JolocomTheme.textStyles.light.subheader }
+            msg={ claimField } />
+          { this.renderInputFields(claimField) }
+        </Block>
+        <Button
```

Comment:
The button here uses the MUI style (rendered blue right now); I think it needs to be harmonized with style from other buttons.